rfcs/0133-git-hashing.md

```diff
@@ -0,0 +1,346 @@
+The practical effect of this discrepancy is that a root file (as opposed to directory) in Nix has permission info, but does not in Git.
+
+If we are trying to convert existing Nix data into Git, this is a problem.
+Assuming we treat "no permission bits" as meaning "non-executable", we will have a partial conversion that will fail on executable files without a parent directory.
+Tricks like always wrapping everything in a directory get around this, but then we have to be careful the directory is exactly as expected when "unwrapping" in the other direction.
+
+For now, we only focus on ingesting data *from* Git *to* Nix, and this side-steps the issue.
+That conversation is total (though not surjective), and so there is no problem for now.
+
+# Alternatives
+[alternatives]: #alternatives
+
+The dependency graph of steps can be sliced to save some for future work.
+For now they are all written together, but during the RFC meetings we will decide which steps (if any) to ratify now, and which steps to save for later.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+None at this time.
+
+# Future work --- Software Heritage
+[future]: #future-work-swh
+
+## Motivation
+
+Software Heritage is a great way to get source code that is no longer found at its original location online.
+It would be really great to use it as a substituter to fetch source code that fails to be fetched the default way.
+
+## Detailed Designed
+
+- **Depends on**: Nix-agnostic content-addressing "stores"
+
+[`GET /api/1/raw/(swhid)/`](https://docs.softwareheritage.org/devel/swh-web/uri-scheme-api.html#get--api-1-raw-(swhid)-)
+is a relatively new Software Heritage API endpoint to get a git object for the given SWHID.
+We can used this to write a Software Heritage store, which will accomplish the task layed out in the previous motivation section.
+
+# Future work --- IPFS
+[future]: #future-work-ipfs
+
+## Motivation
+
+### Binary Distribution
+
+IPFS is a potential solution leveraging content addressing for a P2P CDN.
+It out-of-the-box match's Nix's data model extremely well.
+
+### Source distribution and archival
+
+IPFS also supports git hashing, and so we also provide a good way for people and institutions to "pin" the sources they need, especially if those sources include private ones Software Heritage won't have.
+Finally, per [this IPFS plugin](https://blog.obsidian.systems/software-heritage-bridge/), we can have a 3 way integration between IPFS, Software Heritage, and Nix.
+Data can be directly downloaded from Software Heritage via HTTPS, or indirectly via IPFS.
+
+This would hopefully supercede the direct Software Heritage store outlined above.
+This is because Software Heritage is primarily an archive, and we should be careful not to waste their limitted resources with copious egress bandwidth.
+IPFS can act as a CDN to not put as much load on Software Heritage's servers.
+
+### Not just IPFS
```

Comment:
I initially didn't see the other marked-as-resolved threads, but I think this should be reworded to focus on the content-adressability, and marking IPFS as one possible vehicle/transport for that data.

Reply:
We perhaps should just delete all this text as move it to a draft new RFC :)